community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/NodeRelationshipCache.java

```diff
@@ -410,7 +412,7 @@ public int getCount( long nodeId, Direction direction )
         {
             this.base = base;
             assert chunkSize > 0;
-            this.array = arrayFactory.newDynamicByteArray( chunkSize,
+            this.array = arrayFactory.newDynamicByteArray( 1_000_000,
```

Comment:
Should we also try to calculate number of relationship groups in advance to avoid dynamic array here as well?